### PR TITLE
Fix misc compiler warnings.

### PR DESCRIPTION
**Pull request by ksooo**
Namely:

screen shot 2016-06-18 at 11 05 37

@paxxi for review.

### CODE REVIEW

**Comment by Paxxi** [comment]
Looks good.

File: xbmc/network/WebServer.cpp [hunk 1]
```diff
@@ -974,7 +974,11 @@ int CWebServer::ContentReaderCallback(void *cls, size_t pos, char *buf, int max)
     return -1;
 
   if (g_advancedSettings.CanLogComponent(LOGWEBSERVER))
+#if (MHD_VERSION >= 0x00090200)
```
Comment (Paxxi) [inline xbmc/network/WebServer.cpp]: Semi related, @wsnipex are there any platform that has mhd older than this? Got me thinking if we can get rid of a few ifdefs related to mhd

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
OSX jenkins error is unrelated.

File: xbmc/network/WebServer.cpp (hunk 1 above)
Comment (wsnipex) [inline xbmc/network/WebServer.cpp]: you can drop support for the older versions

Comment (MartijnKaijser) [inline xbmc/network/WebServer.cpp]: @wsnipex what should be minimum version? Our depends currently has 0.9.50

Comment (wsnipex) [inline xbmc/network/WebServer.cpp]: I'd keep the one defined here

**Pull request merged**